Sources/StreamChat/StateLayer/ConnectedUserState.swift

```diff
@@ -17,6 +17,9 @@ import Foundation
             with: .init(
                 userDidChange: { [weak self] in self?.user = $0 })
         )
+        if let user = observer.userObserver.item {
+            self.user = user
+        }
```

Comment:
Planning to do a separate PR for cleaning up some of the initial value part. Since I recently made state properties lazy then now it is important to always check the observer's state and not trust the injected user value which is part of the `init` (we might already have changed the user before state has been loaded, I caught this while writing tests).

Reply:
probably we should also check other places